File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/camera/CameraInfo.java
package com.abubusoft.xenon.camera;

import com.abubusoft.xenon.math.Matrix4x4;
import com.abubusoft.xenon.settings.ProjectionType;

/**
 * Informazione relativi alla camera e alle informazioni ad essa associate.
 * 
 * @author Francesco Benincasa
 * 
 */
public class CameraInfo {

	/**
	 * angolo di visualizzaizione espresso in gradi.
	 */
	public float fieldOfView;

	/**
	 * distanza del piano più vicino rispetto alla camera del frustum
	 */
	public float zNear;

	/**
	 * distanza del piano più lontano del frustum rispetto alla camera
	 */
	public float zFar;

	/**
	 * tipo di proiezione
	 */
	public ProjectionType projection;

	/**
	 * dimensione della camera (valido solo nella proiezione ORTOGONALE).
	 */
	public float frustumSize;

	/**
	 * <p>
	 * viewport sottoforma di matrice.
	 * </p>
	 * 
	 */
	public int[] viewport = new int[16];

	/**
	 * matrice di trasformazione modelView
	 */
	public Matrix4x4 cameraMatrix = new Matrix4x4();

	/**
	 * matrice di proiezione
	 */
	public Matrix4x4 projectionMatrix = new Matrix4x4();

	/**
	 * <p>
	 * Matrice di moltiplicazione projectionMatrix startX cameraMatrix
	 * </p>
	 * 
	 * <p>
	 * Comoda per evitare ad ogni frame la moltiplicazione delle matrici
	 * </p>
	 */
	public Matrix4x4 projection4CameraMatrix = new Matrix4x4();

	/**
	 * <p>Rappresenta 1f/ (zFar - zNear)</p>
	 */
	public float zInverseFrustmDepthFactor;

	/**
	 * Imposta il viewport in termini di width e height, usato
	 * 
	 * @param width
	 * @param height
	 */
	public void setViewport(int width, int height) {
		viewport[2] = width;
		viewport[3] = height;
	}

}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/camera/CameraInfo.kt
package com.abubusoft.xenon.camera

import com.abubusoft.xenon.math.Matrix4x4
import com.abubusoft.xenon.settings.ProjectionType

/**
 * Informazione relativi alla camera e alle informazioni ad essa associate.
 *
 * @author Francesco Benincasa
 */
class CameraInfo {
    /**
     * angolo di visualizzaizione espresso in gradi.
     */
    var fieldOfView = 0f

    /**
     * distanza del piano più vicino rispetto alla camera del frustum
     */
    var zNear = 0f

    /**
     * distanza del piano più lontano del frustum rispetto alla camera
     */
    var zFar = 0f

    /**
     * tipo di proiezione
     */
    var projection: ProjectionType? = null

    /**
     * dimensione della camera (valido solo nella proiezione ORTOGONALE).
     */
    var frustumSize = 0f

    /**
     *
     *
     * viewport sottoforma di matrice.
     *
     *
     */
    var viewport = IntArray(16)

    /**
     * matrice di trasformazione modelView
     */
    var cameraMatrix = Matrix4x4()

    /**
     * matrice di proiezione
     */
    var projectionMatrix = Matrix4x4()

    /**
     *
     *
     * Matrice di moltiplicazione projectionMatrix startX cameraMatrix
     *
     *
     *
     *
     * Comoda per evitare ad ogni frame la moltiplicazione delle matrici
     *
     */
    var projection4CameraMatrix = Matrix4x4()

    /**
     *
     * Rappresenta 1f/ (zFar - zNear)
     */
    var zInverseFrustmDepthFactor = 0f

    /**
     * Imposta il viewport in termini di width e height, usato
     *
     * @param width
     * @param height
     */
    fun setViewport(width: Int, height: Int) {
        viewport[2] = width
        viewport[3] = height
    }
}